interfaces/repo.go

```diff
@@ -46,14 +45,22 @@ type Repository struct {
 
 // NewRepository creates an empty plug repository.
 func NewRepository() *Repository {
-	return &Repository{
+	repo := &Repository{
 		ifaces:    make(map[string]Interface),
 		plugs:     make(map[string]map[string]*Plug),
 		slots:     make(map[string]map[string]*Slot),
 		slotPlugs: make(map[*Slot]map[*Plug]*Connection),
 		plugSlots: make(map[*Plug]map[*Slot]*Connection),
 		backends:  make(map[SecuritySystem]SecurityBackend),
 	}
+
+	return repo
+}
+
+func (r *Repository) SetSanitizePlugSlots() {
+	snap.SanitizePlugsSlots = func(snapInfo *snap.Info) {
+		r.SanitizePlugsSlots(snapInfo)
```

Comment:
This doesn't look right. This is changing global state by a method on a repository instance, setting to its own method. The idea of enabling and disabling the global logic by fiddling with this function is also suspect.

Why do we need the repository for those validations? We know about every interface at runtime without the need to touch the repository. That function should even be changeable I think, but rather an actual function that should always be available.

Reply:
Sorry, I meant the function should _not_ be changeable.

Reply:
Hmm. This function is called from snap.ReadInfo and needs to be a settable callback to avoid import cycle; in snapd it needs to actually sanitize plugs and slots against all interfaces, but in `snap` and `snap-exec` I set it to no-op func not to drag all the interfaces in (plus, it doesn't make sense to sanitize at this point).

To summarize, I'm not sure how to change this aspect to satisfy both cases - other than by having two ReadInfo functions (with- and without- sanitize), which I think is a bad idea.

NB, I got rid of the of the SetSanitizePlugSlots setter and do not depend on the repository anymore.